bot/converters.py

```diff
@@ -495,22 +495,55 @@ async def convert(self, ctx: Context, argument: str) -> int:
         return duration
 
 
-class UserMentionOrID(UserConverter):
+def _is_an_unambiguous_user_argument(argument: str) -> bool:
+    """Check if the provided argument is a user mention, user id, or username."""
+    has_id_or_mention = bool(IDConverter()._get_id_match(argument) or RE_USER_MENTION.match(argument))
+
+    if not has_id_or_mention:
+        if argument[0] == '@':
+            argument = argument[1:]
+
+        # Check to see if the author passed a username (a discriminator exists)
+        if len(argument) > 5 and argument[-5] == '#':
+            return True
+
+    return has_id_or_mention
```

Comment:
```suggestion
    if has_id_or_mention:
        return has_id_or_mention
    
    if argument[0] == '@':
        argument = argument[1:]

    # Check to see if the author passed a username (a discriminator exists)
    if len(argument) > 5 and argument[-5] == '#':
        return True
```

Reply:
That wouldn't work as-is (returns None sometimes) but I could instead use this to make the flow better

```py
    has_id_or_mention = ...

    # Check to see if the author passed a username (a discriminator exists)
    argument = argument.remove_prefix('@')
    has_username = len(argument) > 5 and argument[-5] == '#'

    return has_id_or_mention or has_username
```